### PR TITLE
Add plugin implementation

### PR DESCRIPTION
**Pull request by rpjayasekara**
Added Ballerina Gradle plugin implementation and plugin publish logic

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,48 @@
+plugins {
+    id 'groovy'
+    id 'java-gradle-plugin'
+    id 'maven-publish'
+    id 'com.gradle.plugin-publish' version '0.14.0'
+}
+
+group 'io.ballerina'
+version '0.0.1'
```
Comment (keizer619) [inline build.gradle:9]: @rpjayasekara 
Shall we move these versions into `gradle.properties` file?

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        maven {
+            name = 'localPluginRepository'
+            url = '../local-plugin-repository'
+        }
+    }
+}
+
+test {
+    useJUnitPlatform()
+}
```
Comment (keizer619) [inline build.gradle:48]: @rpjayasekara Lets include a new line in the end

File: src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy [hunk 3]
```diff
@@ -0,0 +1,271 @@
+package io.ballerina.plugin
```
Comment (keizer619) [inline src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy:1]: @rpjayasekara Shall we include a License header for groovy source?

This a sample License header used for Java
https://github.com/ballerina-platform/ballerina-lang/blob/master/language-server/modules/langserver-commons/src/main/java/org/ballerinalang/langserver/commons/CodeActionContext.java#L1

File: build.gradle (hunk 1 above)
Comment (rpjayasekara) [reply build.gradle:9]: okay

File: build.gradle (hunk 2 above)
Comment (rpjayasekara) [reply build.gradle:48]: okay

File: src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy (hunk 3 above)
Comment (rpjayasekara) [reply src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy:1]: Shall we use the same license header?

Comment (keizer619) [reply src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy:1]: +1

**Comment by keizer619** [comment]
@rpjayasekara Lets update the `README.md` with plugin details and with sample usage

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+name: Release
```
Comment (keizer619) [inline .github/workflows/release.yml:1]: @rpjayasekara Shall we inlcude PR workflow and master build workflows as well?

**Comment by keizer619** [comment]
Usually for commit messages we starts a capital letter and 
"add build main branch workflow" should be like "Add build main branch workflow"

You can follow this guide
https://chris.beams.io/posts/git-commit/

**Review by keizer619 — APPROVED** [review]

**Pull request merged**